src/Language/DifferentialDatalog/Debugger/DebugEventParser.hs

```diff
@@ -0,0 +1,160 @@
+{-
+Copyright (c) 2018-2020 VMware, Inc.
+SPDX-License-Identifier: MIT
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+-}
+
+module Language.DifferentialDatalog.Debugger.DebugEventParser (
+    eventsParser) where
+
+import qualified Text.Parsec.Token as T
+import Text.Parsec
+import Text.Parsec.Language
+import Data.Functor.Identity
+
+data CollectionType  = Vector deriving (Show)
+data Operator = OpMap | OpAggregate | OpCondition | OpJoin
+                | OpAntijoin | OpInspect | OpUndefined deriving (Show)
+
+data Events = Events { events :: [Event]} deriving (Show)
+
+data Event = DebugEvent { operatorId :: (Integer, Integer, Integer)
+                         , weight :: Integer
```

Comment:
Unfortunately, all field names in Haskell are functions in the global namespace, so fields like `weight` and `timestamp` will likely introduce name clashes.  I therefore use qualified field names like `evtWeight`, `evtTimestamp`, etc.